lib/puppetserver/ca/action/migrate.rb

```diff
@@ -0,0 +1,95 @@
+require 'puppetserver/ca/utils/cli_parsing'
+require 'puppetserver/ca/utils/file_system'
+require 'puppetserver/ca/utils/http_client'
+
+module Puppetserver
+  module Ca
+    module Action
+      class Migrate
+        include Puppetserver::Ca::Utils
+        PUPPETSERVER_CA_DIR = '/etc/puppetlabs/puppetserver/ca'
+
+        SUMMARY = "Migrate the CA directory /etc/puppetlabs/puppetserver/ca"
+        BANNER = <<-BANNER
+Usage:
+  puppetserver ca migrate [--help]
+  puppetserver ca migrate [--config PATH]
+
+Description:
+  Migrate an existing CA directory in the puppet directory to puppetserver. This is for
+  upgrading from Puppet platform 6.x to Puppet 7. Use the currently configured
+  puppet.conf file in your installation, or supply one using the `--config` flag.
+Options:
+BANNER
+
+        def initialize(logger)
+          @logger = logger
+        end
+
+        def run(input)
+          config_path = input['config']
```

Comment:
How is the default config path handled, i.e. when the `--config` flag isn't passed? We might want a test for that?

Reply:
There is a test for when the `--config` flag is not passed, in the spec for "no required arguments".

Reply:
Also fwiw, the `Config::Puppet` class handles the case where `config_path` is nil: https://github.com/puppetlabs/puppetserver-ca-cli/blob/master/lib/puppetserver/ca/config/puppet.rb#L35-L41

Reply:
I saw the test for when `--config` isn't passed, but it doesn't test that a default is used, just that it's nil. But I guess that is handled elsewhere.